account_banking_pain_base/models/account_payment_order.py

```diff
@@ -480,7 +459,31 @@ def generate_address_block(self, parent_node, partner, gen_args):
                     70,
                     gen_args=gen_args,
                 )
-
+            if (
+                gen_args.get("pain_flavor").startswith("pain.001.001.")
+                or gen_args.get("pain_flavor").startswith("pain.008.001.")
+                and (partner.zip or partner.city)
+            ):
+                adrline2 = etree.SubElement(postal_address, "AdrLine")
+                if partner.zip:
+                    val = self._prepare_field(
+                        "zip",
+                        "partner.zip",
+                        {"partner": partner},
+                        70,
+                        gen_args=gen_args,
+                    )
+                else:
+                    val = ""
+                if partner.city:
+                    val += " " + self._prepare_field(
+                        "city",
+                        "partner.city",
+                        {"partner": partner},
+                        70,
+                        gen_args=gen_args,
+                    )
+                adrline2.text = val
```

Comment:
You might want to use a list here to prevent a leading space:

```
val = []
if partner.zip:
    val.append(...)
if partner.city:
    val.append(...)
adrline2.text = " ".join(val)
```